src/components/destination_list.js

```diff
@@ -8,27 +8,50 @@ import NoItems from "./items/no_items";
 import SelectedItem from "./items/selected_item";
 import SelectionStatus from "./selection_status/selection_status";
 import { groupItems } from "./item_grouping_util";
+import Search from "./search/search";
 
 const DestinationList = ({
   selectionStatusRenderer,
   selectedIds,
   clearAll,
   messages,
-  selectedItems,
   itemHeight,
   height,
   unselectItems,
   selectedItemRenderer,
   noItemsRenderer,
-  withGrouping
+  withGrouping,
+
+  showRightSearch,
+  searchIcon,
+  selectedItems,
+  serchRightValue,
+  serchRightValueChange,
+  filterRightSearch
 }) => {
   const SelectionStatusRenderer = selectionStatusRenderer;
+
+  let filterSelectitems = selectedItems;
+  if (serchRightValue && serchRightValue.length) {
```

Comment:
This is not logic for the component, please follow the components' guidelines.
This should be in the state management level.